src/app/reporting/SynchronizedReportSchedulerImpl.cpp

```diff
@@ -206,12 +201,14 @@ void SynchronizedReportSchedulerImpl::TimerFired()
 
     if (firedEarly)
     {
+        // If we fired the timer early, we need to recalculate the next report timeout and reschedule the report
```

Comment:
"reschedule the report for a time when we should have a handler that's ready to report"?  Or for some other reason?  _Why_ do we need to do this?  That's the question this comment leaves unanswered.

Reply:
The reason is that when we fire early, we essentially say: "my timer disagrees with the monotonic timestamp but I will trust the timer so let's ignore the min timestamp". But since we use a common timer in this implementation, we don't know which handler should be reportable now and we don't want to systematically ignore min intervals.

(I wrote something along those lines in the update doc PR).